app/controllers/incomeSources/add/AddIncomeSourceStartDateController.scala

```diff
@@ -0,0 +1,260 @@
+      if (isAgent) itvcErrorHandlerAgent.showInternalServerError()
+      else itvcErrorHandler.showInternalServerError()
+  }
+
+  private def getCallsAndMessagesKeyPrefix(isAgent: Boolean, incomeSourceType: IncomeSourceType): (Call, Call, Call, String) = {
+
+    (isAgent, incomeSourceType) match {
+      case (false, SoleTraderBusiness) =>
+        (
+          controllers.incomeSources.add.routes.AddBusinessNameController.show(),
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitSoleTraderBusiness,
+          controllers.incomeSources.add.routes.AddBusinessStartDateCheckController.show(),
+          soleTraderBusinessMessagesPrefix
+        )
+      case (false, UKProperty) =>
+        (
+          controllers.incomeSources.add.routes.AddIncomeSourceController.show(),
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitUKProperty,
+          controllers.incomeSources.add.routes.CheckUKPropertyStartDateController.show(),
+          uKPropertyMessagesPrefix
+        )
+      case (false, ForeignProperty) =>
+        (
+          controllers.incomeSources.add.routes.AddIncomeSourceController.show(),
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitForeignProperty,
+          controllers.incomeSources.add.routes.ForeignPropertyStartDateCheckController.show(),
+          foreignPropertyMessagesPrefix
+        )
+      case (true, SoleTraderBusiness) =>
+        (
+          controllers.incomeSources.add.routes.AddBusinessNameController.showAgent(),
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitSoleTraderBusinessAgent,
+          controllers.incomeSources.add.routes.AddBusinessStartDateCheckController.showAgent(),
+          soleTraderBusinessMessagesPrefix
+        )
+      case (true, UKProperty) =>
+        (
+          controllers.incomeSources.add.routes.AddIncomeSourceController.showAgent(),
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitUKPropertyAgent,
+          controllers.incomeSources.add.routes.CheckUKPropertyStartDateController.showAgent(),
+          uKPropertyMessagesPrefix
+        )
+      case (true, ForeignProperty) =>
+        (
+          controllers.incomeSources.add.routes.AddIncomeSourceController.showAgent(),
+          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitForeignPropertyAgent,
+          controllers.incomeSources.add.routes.ForeignPropertyStartDateCheckController.showAgent(),
+          foreignPropertyMessagesPrefix
+        )
+    }
+  }
+
+  private lazy val soleTraderBusinessMessagesPrefix = "add-business-start-date"
+  private lazy val uKPropertyMessagesPrefix = "incomeSources.add.UKPropertyStartDate"
+  private lazy val foreignPropertyMessagesPrefix = "incomeSources.add.foreignProperty.startDate"
+
```

Comment:
you can move this under 
/Users/hmrc/devcore/itvc/income-tax-view-change-frontend/app/enums